src/Analyzer.Utilities/CSharp/ITypeSymbolExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Analyzer.Utilities
+{
+    public static class ITypeSymbolExtensions
+    {
+        public static bool IsAssignableTo(this ITypeSymbol fromSymbol, ITypeSymbol toSymbol, Compilation compilation)
```

Comment:
+1. There should be no need to write language-specific rules just to check assignability. This should be handled polymorphically by core API.